flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -386,10 +386,10 @@ private ClusterSpecification createClusterSpecification(Configuration configurat
 		}
 
 		// JobManager Memory
-		final int jobManagerMemoryMB = MemorySize.parse(configuration.getString(JobManagerOptions.JOB_MANAGER_HEAP_MEMORY)).getMebiBytes();
+		final int jobManagerMemoryMB = MemorySize.parse(configuration.getString(JobManagerOptions.JOB_MANAGER_HEAP_MEMORY), MemorySize.MemoryUnit.MEGA_BYTES).getMebiBytes();
```

Comment:
Was there a reason against making `JOB_MANAGER_HEAP_MEMORY_MB` a deprecated key for `JOB_MANAGER_HEAP_MEMORY` when you worked on FLINK-6469? That is:
```
public static final ConfigOption<String> JOB_MANAGER_HEAP_MEMORY =
		key("jobmanager.heap.size")
			.defaultValue("1024m")
			.withDeprecatedKeys("jobmanager.heap.mb")
		    ...
```
I am not sure what happens if someone only configures `jobmanager.heap.mb` in `flink-conf.yaml`. Will the value be respected for all deployment modes?

Same holds for the TaskManager config key.

Reply:
in FLINK-6469, in order to config the jm's memory with unit I introduced a new key and deprecated `jobmanager.heap.mb`.

* in flink codebase(except shell script) I have removed all the place used `JOB_MANAGER_HEAP_MEMORY_MB` and `jobmanager.heap.mb`, so it will not be used.
* in shell (`config.sh`) the old key `jobmanager.heap.mb` also be supported if the new key `jobmanager.heap.size` can not be read, so it still be supported.

Reply:
Would it hurt to add `.withDeprecatedKeys("jobmanager.heap.mb")`? In case someone does not use the scripts to start the cluster.

Reply:
I think the problem here would be that we would need to get back to MB as a default unit for this option. Or am I wrong here? The whole idea for introducing the new option was to switch to bytes as a default one, if none provided.

Reply:
@dawidwys As I understand, backwards compatibility should be prioritized. See https://github.com/apache/flink/pull/5448#issuecomment-394296226

> We need to add an additional MemoryUnit.parse() method that takes the "default" unit, so that we parse the old heap sizes such that they are in MB if nothing else is specified.

Reply:
Agreed, missed it previously. Unfortunately `withDeprecatedKeys` won't solve it though. Either we should revert the option (and use the mb versions), or anywhere we use the new option, we should also check for the old one.

Reply:
@dawidwys Why is it not enough?

> When obtaining a value from the configuration via Configuration.getValue(ConfigOption), the deprecated keys will be checked in the order provided to this method. The first key for which a value is found will be used - that value will be returned.

Reply:
Imagine a case when previously we had in configuration:

    jobmanager.heap.mb: 1024

which meant 1024 MB, in the new code we will pass 1024 as value for `jobmanager.heap.size`, which was introduced solely to treat numbers without units as bytes. Therefore it will be parsed as 1024 bytes rather than 1024 MB.

If we switch the default unit to MB, there is no point for having a new `ConfigOption`.

Reply:
to @GJL we can not use `.withDeprecatedKeys("jobmanager.heap.mb")` because `jobmanager.heap.size` and `jobmanager.heap.mb` has different meaning. The former can use different unit such **1g** but the latter can just measure with **MB**.

to @dawidwys and @GJL , now the `jobmanager.heap.mb` just used in config file, and can be calculated accurately, this is used for backwards compatibility, but in the project, it is useless, all the place can be replaced with `jobmanager.heap.size`, and the key in the code could not been exposed to the user?

Reply:
introduce a new config key is used to make the `jobmanager.heap.mb` backwards compatibility in flink config file (config.sh can calculate it accurately). And user can specify the unit for the value of the key `jobmanager.heap.size` .

So if we remove anything about `JOB_MANAGER_HEAP_MEMORY_MB ` in Java and Scala code, is there any problem?

Reply:
I think we cannot do this as users might not use the `config.sh`. At the same time we think we really should keep backwards compatibility, as otherwise there might be complaints.  We've discussed this with @GJL offline and we think the way to go right now would be to check everywhere we use the new option to check for the old one as well if the new one was not set.

That is:
* `org.apache.flink.client.deployment.ClusterSpecification#fromConfiguration`
* `org.apache.flink.yarn.YarnResourceManager#YarnResourceManager`
* `org.apache.flink.yarn.cli.FlinkYarnSessionCli#createClusterSpecification`

Probably we should wrap it in some utility method.

Also for the yarn command line we should keep the old behaviour. This means we should add "m" suffix in `org.apache.flink.yarn.cli.FlinkYarnSessionCli#applyCommandLineOptionsToConfiguration`